src/tools.tsx

```diff
@@ -17,7 +19,10 @@ export function withSuperJSONProps<P>(
 
     const { json, meta } = SuperJSON.serialize(result.props);
 
-    const props: SuperJSONResult = { json };
+    const props: SuperJSONProps = {
+      json,
+      _superjson: true,
```

Comment:
This adds unnecessary heft to the page's payload. Why can't we stay with `json`?

Reply:
Because each page has the HOC now, just checking presence of `props.json` is prone to bugs. We need to know absolutely that it's a superjson payload.